### PR TITLE
feat: [CDS:89201]: Add documentation for pipeline.executionMode

### PR DESCRIPTION
**Pull request by ayushitiwari024**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue: No documentation available for pipeline.executionMode. Added the documentation and the screenshot for the same.
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 1]
```diff
@@ -746,6 +746,25 @@ https://app.harness.io/ng/#/account/12345678910/cd/orgs/default/projects/CD_Docs
 The expression \<+pipeline.execution.Url> has been deprecated.
 :::
 
+
+
+### \<+pipeline.executionMode>
+
+The execution mode  of the pipeline. This will tell the execution mode(i.e NORMAL, POST_EXECUTION_ROLLBACK) of the pipeline.
```
Comment (SudarshanPurohit) [inline docs/platform/variables-and-expressions/harness-variables.md:753]: I think we need to expand a bit on what "execution mode" means. Users don't know what they are (unless this is defined somewhere else and we can link to it). 

So, maybe an additional sentence saying, 

```
The Execution Mode is a string that can have two possible values currently: 
* "NORMAL": a normal execution, which could have been either successful or failed. \
* "POST_EXECUTION_ROLLBACK": this was a Post Execution Rollback (add link to Post Prod Rollback here) ) triggered in order to roll back a specific service. 
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 2]
```diff
@@ -746,6 +746,25 @@ https://app.harness.io/ng/#/account/12345678910/cd/orgs/default/projects/CD_Docs
...
+
+![](./static/execution-mode-expression.png)
+
+We can use the expression `<+pipeline.executionMode>` under conditional execution.
```
Comment (SudarshanPurohit) [inline docs/platform/variables-and-expressions/harness-variables.md:757]: Probably add an example use case? I like how you've added the screenshots, and additionally something like "conditional execution to ensure a step only runs when a Post Execution Rollback has been triggered."

File: docs/platform/variables-and-expressions/harness-variables.md (hunk 1 above)
Comment (ayushitiwari024) [reply docs/platform/variables-and-expressions/harness-variables.md:753]: well, it has more than 2 values so do we need to specify the meaning of all the values?

Comment (ayushitiwari024) [reply docs/platform/variables-and-expressions/harness-variables.md]: @sahil please review my PR and also review the changes for Post_Execution_Rollback and Post_Deployment_Rollback. On UI for documentation, we have Post deployment Rollback page but from backend, we have post_execution_rollback as our enum

**Review by prasadsatamharness — DISMISSED** [review]
This looks good, thank you @ayushitiwari024

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 3]
```diff
@@ -746,6 +746,32 @@ https://app.harness.io/ng/#/account/12345678910/cd/orgs/default/projects/CD_Docs
 The expression \<+pipeline.execution.Url> has been deprecated.
 :::
 
+
+
+### \<+pipeline.executionMode>
+
+The execution mode  of the pipeline. This will tell the execution mode(i.e NORMAL, POST_EXECUTION_ROLLBACK, PIPELINE_ROLLBACK) of the pipeline.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:753]:
```suggestion
The execution mode of the pipeline (`NORMAL`, `POST_EXECUTION_ROLLBACK`, or `PIPELINE_ROLLBACK`.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 4]
```diff
@@ -746,6 +746,32 @@ https://app.harness.io/ng/#/account/12345678910/cd/orgs/default/projects/CD_Docs
...
+
+**Types of Execution Mode:**
+1) Normal Execution: A execution which could either be successful or failed.
+
+1) [Post deployment Rollback[POST_EXECUTION_ROLLBACK]](../../continuous-delivery/manage-deployments/rollback-deployments.md)
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:758]:
```suggestion
2) [Post deployment Rollback[`POST_EXECUTION_ROLLBACK`]](../../continuous-delivery/manage-deployments/rollback-deployments.md)
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 5]
```diff
@@ -746,6 +746,32 @@ https://app.harness.io/ng/#/account/12345678910/cd/orgs/default/projects/CD_Docs
...
+
+2) [Rollback pipelines[PIPELINE_ROLLBACK]](../../platform/pipelines/define-a-failure-strategy-for-pipelines.md)
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:760]:
```suggestion
3) [Rollback pipelines[`PIPELINE_ROLLBACK`]](../../platform/pipelines/define-a-failure-strategy-for-pipelines.md)
```

**Review by brian-f-harness — APPROVED** [review]

File: docs/platform/variables-and-expressions/harness-variables.md (hunk 3 above)
Comment (ayushitiwari024) [reply docs/platform/variables-and-expressions/harness-variables.md]: addressed

**Review by brian-f-harness — APPROVED** [review]

**Review by prasadsatamharness — APPROVED** [review]

**Pull request merged**